fvm/fvm_test.go

```diff
@@ -1825,6 +1825,331 @@ func TestSignatureVerification(t *testing.T) {
+							signatures = append(signatures, sig)
+						}
+
+						script := fvm.Script(code).WithArguments(
+							jsoncdc.MustEncode(cadence.Array{
+								Values: signatures,
+								ArrayType: cadence.VariableSizedArrayType{
+									ElementType: cadence.VariableSizedArrayType{
+										ElementType: cadence.UInt8Type{},
+									},
+								},
+							}),
+						)
+
+						err = vm.Run(ctx, script, view, programs)
+						switch signatureAlgorithm.algorithm {
+						case crypto.BLSBLS12381:
+							assert.NoError(t, err)
+							assert.NoError(t, script.Err)
+
+							aggSk, err := crypto.AggregateBLSPrivateKeys(sks)
+							require.NoError(t, err)
+							expectedSig, err := aggSk.Sign(input, hasher)
+							require.NoError(t, err)
+
+							assert.Equal(t, testutil.BytesToCadenceArray(expectedSig), script.Value)
+						case crypto.ECDSAP256, crypto.ECDSASecp256k1:
+							assert.NoError(t, err)
+							assert.Error(t, script.Err)
+						}
+					})
+				},
+			))
+		})
+		t.Run(signatureAlgorithm.name, func(t *testing.T) {
+			t.Run("aggregateBLSPublicKeys", newVMTest().run(
+				func(
+					t *testing.T,
+					vm *fvm.VirtualMachine,
+					chain flow.Chain,
+					ctx fvm.Context,
+					view state.View,
+					programs *programs.Programs,
+				) {
+
+					pkNum := 100
+
+					code := []byte(
+						fmt.Sprintf(
+							`
+								import Crypto
+		
+								pub fun main(
+									publicKeys: [[UInt8]]
+								): [UInt8] {
+									let pks: [PublicKey] = []
+									for pk in publicKeys {
+										pks.append(PublicKey(
+											publicKey: pk, 
+											signatureAlgorithm: SignatureAlgorithm.%s
+										))
+									}
+									return AggregateBLSPublicKeys(pks).publicKey
+								}
+								`,
+							signatureAlgorithm.name,
+						),
+					)
+
+					t.Run("", func(t *testing.T) {
+						pks := make([]crypto.PublicKey, 0, pkNum)
+						sks := make([]crypto.PrivateKey, 0, pkNum)
+						publicKeys := make([]cadence.Value, 0, pkNum)
+
+						// create the signatures
+						for i := 0; i < pkNum; i++ {
+							sk := randomSK(t, seed)
+							sks = append(sks, sk)
+							pks = append(pks, sk.PublicKey())
+							publicKeys = append(
+								publicKeys,
+								testutil.BytesToCadenceArray(sk.PublicKey().Encode()),
+							)
+						}
+
+						script := fvm.Script(code).WithArguments(
+							jsoncdc.MustEncode(cadence.Array{
+								Values: publicKeys,
+								ArrayType: cadence.VariableSizedArrayType{
+									ElementType: cadence.VariableSizedArrayType{
+										ElementType: cadence.UInt8Type{},
+									},
+								},
+							}),
+						)
+
+						err = vm.Run(ctx, script, view, programs)
+						switch signatureAlgorithm.algorithm {
+						case crypto.BLSBLS12381:
+							assert.NoError(t, err)
+							assert.NoError(t, script.Err)
+
+							aggSk, err := crypto.AggregateBLSPrivateKeys(sks)
+							require.NoError(t, err)
+							expectedPk := aggSk.PublicKey()
+							require.NoError(t, err)
+
```

Comment:
What is implemented is correct, though we can make a shortcut. 
The tests in the crypto library are made differently because they are testing the arithmetic relationships, while here we're just testing the FVM functions are wired correctly.
```suggestion
							expectedPk, err := crypto.AggregateBLSPublicKeys(sks)
							require.NoError(t, err)
```